test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -967,6 +968,23 @@ public XContentBuilder shuffleXContent(XContentParser parser, boolean prettyPrin
         return xContentBuilder.map(shuffledMap);
     }
 
+    /**
+     * Returns the bytes that represent the XContent output of the provided {@link ToXContent} object, using the provided
+     * {@link XContentType}.
+     * Adds a random field value, inner object or array into each json object.
+     */
+    protected final BytesReference toShuffledXContentAndInsertRandomFields(ToXContent toXContent, XContentType xContentType,
+            ToXContent.Params params, boolean humanReadable, String... exceptFieldNames) throws IOException {
+        BytesReference originalBytes = toShuffledXContent(toXContent, xContentType, params, humanReadable, exceptFieldNames);
+        BytesReference mutated;
+        if (randomBoolean()) {
+            mutated = insertRandomFields(xContentType, originalBytes, null, random());
```

Comment:
given that in most cases we have two different test methods for this (with and without random fields), maybe we should not add this new method and rather do it like we did up until now? That said, I think I was not very diligent on this :)